packages/openai/src/chat.ts

```diff
@@ -124,14 +124,14 @@ export class OpenAIChatModel implements IChatModel<ChatCompletionCreateParams> {
     const messages = await memory.values();
 
     if (options.system) {
-      messages.push(options.system);
+      messages.unshift(options.system);
     }
 
     try {
       const completion = await this._openai.chat.completions.create({
         ...this.options.requestOptions,
         ...options.request,
-        model: 'endpoint' in this.options ? '' : this.options.model,
+        model: this.options.model,
```

Comment:
@aacebo do you know why we need this? It wasn't playing well with my AOAI configs

Reply:
that needs to be present for AOAI types, ie if 'endpoints' is in the options they are for AOAI, which you shouldn't pass a `model`

Reply:
Removed it